### PR TITLE
Adding awscreds

### PR DESCRIPTION
**Pull request by giller13**
Hey, adding option to be able to specify key and secret, or profile when calling CF funcs.

### CODE REVIEW

**Comment by giller13** [comment]
@andyday Hey, sorry for the broken CI, but CircleCI is suppressing the export of the AWS env vars on the PR.

File: index.js [hunk 1]
```diff
@@ -75,10 +75,22 @@ var Promise = require('bluebird'),
...
+        if (awscreds.accessKeyId && awscreds.secretAccessKey) {
+            AWS.config.update({"accessKeyId": awscreds.accessKeyId, "secretAccessKey": awscreds.secretAccessKey});
+        } else if (awscreds.profile) {
+            AWS.config.credentials = new AWS.SharedIniFileCredentials({profile: awscreds.profile});
+        }
+        if(awscreds.region) {
+            AWS.config.update({"region": awscreds.region });
+        }
+    }
+
```
Comment (andyday) [inline index.js]: would prefer naming this awsConfig and adding it as part of the opts object.  below we set opts from name if it is an object.  calling Cfn with name and template is the short form.  anything beyond this just passes in an opts object as the only param.  also would like the awsConfig object to be used in 2 ways:

```javascript
awsConfig = opts.awsConfig;
if (awsConfig && awsConfig.profile) {
    AWS.config.credentials = new AWS.SharedIniFileCredentials({ profile: awsConfig.profile });
} else if (awsConfig) {
    AWS.config.update(awsConfig);
}
```

**Comment by andyday** [comment]
no worries on the build.  i'll be changing this to use mocks sometime soon...

**Comment by andyday** [comment]
this has been covered by #2 ...

**Pull request closed without merging**